src/main/java/hudson/plugins/s3/S3BucketPublisher.java

```diff
@@ -69,10 +69,16 @@ public S3BucketPublisher(String profileName, List<Entry> entries, List<MetadataP
 
         this.dontWaitForConcurrentBuildCompletion = dontWaitForConcurrentBuildCompletion;
         this.consoleLogLevel = parseLevel(consoleLogLevel);
-        this.pluginFailureResultConstraint = Result.fromString(pluginFailureResultConstraint);
+        if (pluginFailureResultConstraint == null) {
+            this.pluginFailureResultConstraint = Result.FAILURE;
+        } else {
+            this.pluginFailureResultConstraint = Result.fromString(pluginFailureResultConstraint);
+        }
     }
 
     private Level parseLevel(String lvl) {
+        if (lvl == null)
+            lvl = "";
```

Comment:
I actually didn't test that, I just added this conversion to be safe. :)

I'll test the other way now.